### ISSUE TITLE
Return pydantic object from function

### ISSUE THREAD

**Issue by benlindsay** [LukasMasuch/streamlit-pydantic#3]
Thanks for making this repo @LukasMasuch! Excited to start playing with it. Would you be open to a PR to modify the API so that `sd.pydantic_input()` returns the instantiated pydantic object, or None if the inputs are invalid? If so I can play around with that

**Comment by LukasMasuch** [comment]
@benlindsay Yes, totally open for PRs on this library :) 

I have actually a small unreleased snippet based on your `pydantic_form` suggestion from Twitter, that returns the instantiated object (see the code below - just run it via `streamlit run`). The nice thing is that the `pydantic_form` also contains data validation out-of-the-box once submit is clicked. Returning the instantiated Pydantic object in `pydantic_input` is also possible, but slightly more complicated because of the data validation since it does not contain the submit button (validate data on any change or instantiated invalidated Pydantic objects). From an API perspective: `pydantic_input` could be the more flexible lower-level building block, while a new function like `pydantic_form` is the faster and more compact way to build forms based on Pydantic classes.

```python
from typing import Optional, Type

import streamlit as st
from pydantic import BaseModel, HttpUrl, ValidationError, parse_obj_as

import streamlit_pydantic as sp


def pydantic_form(
    key: str, input_class: Type[BaseModel], submit_label: str = "Submit"
) -> Optional[BaseModel]:
    with st.form(key=key):
        session_data_key = key + "-data"
        sp.pydantic_input(input_class, session_data_key)
        submit_button = st.form_submit_button(label=submit_label)

        if submit_button:
            try:
                return parse_obj_as(input_class, st.session_state[session_data_key])
            except ValidationError as ex:
                st.error(ex)
                return None
    return None


class ExampleModel(BaseModel):
    url: HttpUrl
    some_number: int
    some_boolean: bool


data = pydantic_form(key="my_form", input_class=ExampleModel)
if data:
    st.json(data.json())
```

If this already solves your need, I can also just release a new version with a `pydantic_form` method :)

**Comment by benlindsay** [comment]
Yeah, that would work perfectly for me! I anticipate only using the form version. I could see others wanting a non-form option that returns an object too, but your proposal sounds like a good compromise that avoids the potential 
performance hit of validating/parsing on every change

**Comment by LukasMasuch** [comment]
@benlindsay I just released version 0.2.0 which contains the proposed `pydantic_form` method. There is now also a playground to try out the examples: https://share.streamlit.io/lukasmasuch/streamlit-pydantic/main/playground/playground_app.py
